### PR TITLE
[v2.2 Audit Fix] Properly account for settlementFee in Vault checkpoint

### PR DESCRIPTION
**Pull request by kbrizzle**
Resolves: https://github.com/sherlock-audit/2024-02-perennial-v2-3-judging/issues/25.

Also [fixes additional issue](https://github.com/equilibria-xyz/perennial-v2/pull/304/files#diff-e919af16c562488455a0addac80be085ca0e3fe833b9d3f3e00a41192a87d8b9R167) where the local settlement fee was being undercharged due to it being rounded down instead of up.

### CODE REVIEW

File: packages/perennial-vault/contracts/types/Checkpoint.sol [hunk 1]
```diff
@@ -113,94 +114,65 @@ library CheckpointLib {
...
+        return (self.shares.isZero() ? shares : _toAssets(self, shares)).unsafeSub(_settlementFeePerOrder(self));
     }
 
     /// @notice Converts a given amount of shares to assets with checkpoint in the global context
     /// @dev Dev used in limit calculations when a non-historical settlement fee must be used
     /// @param shares Number of shares to convert to shares
     /// @param settlementFee Custom settlement fee
     /// @return Amount of assets for the given shares at checkpoint
-    function toAssets(Checkpoint memory self, UFixed6 shares, UFixed6 settlementFee) internal pure returns (UFixed6) {
+    function toAssetsCustom(Checkpoint memory self, UFixed6 shares, UFixed6 settlementFee) internal pure returns (UFixed6) {
```
Comment (arjun-io) [inline packages/perennial-vault/contracts/types/Checkpoint.sol:135]: nit: maybe `toAssetsLimits` or something similar? "Custom" is a bit vague

**Review by arjun-io — APPROVED** [review]
Small nit on naming, LGTM overall

**Review by EdNoepel — APPROVED** [review]
Agree with @arjun-io 's comment regarding naming.

File: packages/perennial-vault/contracts/types/Checkpoint.sol (hunk 1 above)
Comment (kbrizzle) [reply packages/perennial-vault/contracts/types/Checkpoint.sol]: Ended up changing the name back to the original `toAssets`.

**Pull request merged**